stream/src/public_key/handshake.rs

```diff
@@ -198,6 +202,119 @@ impl<C: PublicKey> EncodeSize for Signed<C> {
     }
 }
 
+/// Key confirmation message used during the handshake process.
+///
+/// This struct contains cryptographic proof that a party can correctly derive
+/// the shared secret from the Diffie-Hellman exchange. It prevents attacks where
+/// an adversary might forward handshake messages without actually knowing the
+/// corresponding private keys.
+pub struct KeyConfirmation {
+    /// AEAD tag of the encrypted proof demonstrating knowledge of the shared secret.
+    tag: Tag<ChaCha20Poly1305>,
+}
+
+impl KeyConfirmation {
+    /// Create a new key confirmation using the provided cipher and handshake transcript.
+    ///
+    /// The confirmation encrypts the handshake transcript to demonstrate
+    /// knowledge of the shared secret and bind the confirmation to the entire handshake exchange.
+    pub fn create(mut cipher: ChaCha20Poly1305, transcript: &[u8]) -> Result<Self, Error> {
```

Comment:
Can you add a comment that this `cipher` isn't the one used for ongoing stream encryption (as noted in another comment, the fear is that it may use the same nonce multiple times and IMO would be good to call out why that can't be the case).